### PR TITLE
Find in current file command

### LINKED ISSUE #14836: Add a find
- VSCode Version:
- OS Version:

Steps to Reproduce:

1. 
2.

### PR DESCRIPTION
**Pull request by ssigwart**
This PR fixes #14836.

I did not read all of the comments on that issue since it's extensive, but I implemented @alexdima's suggestion in https://github.com/microsoft/vscode/issues/14836#issuecomment-258781578.

This adds a way to find occurrences in the current file.  To test, you can open the find widget, then click the new icon to find in the file.  I wasn't sure what icon to use.  The two I thought fit best were `file` or `search`.
image
image

I went with `search`, but I'm open to suggestions.

### CODE REVIEW

**Comment by ssigwart** [comment]
@rebornix, should I rebase to fix the conflict?

**Comment by meche-gh** [comment]
I agree that the icon choice here is pretty hard.
There were two feature suggestions in the original feature:

1. Open all results for 1 file in new search editor.
2. **Select all occurences in current document.**

Based on that:

- Ideal icon for the other feature (`#1`) has to be `new-file`.
- Thus we can't use a "file" related codicon, unless its a new `file-search` icon. 📄<sub>🔍</sub>

I did a writeup at https://github.com/microsoft/vscode/issues/14836#issuecomment-1383112522 suggesting a full UX path for this component.
To summarize:
- I landed on `expand-all` or `plus` for this icon.
- To bring this full-circle, I suggested logic to swap it with a `search-clear-results` icon that selects only the first visible search result again and re-enables the arrow icons.

**Review by zm-cttae — CHANGES_REQUESTED** [review]
~~I think this PR should be retooled - we want to expose `SelectAllMatchesAction` as a button. "Find in current file" would be a new PR that opens a new search editor by shelling out to the  "Search" pane on the left.~~

**Comment by ssigwart** [comment]
Thanks for taking the time to review this @meche-gh. However, I'm not going to make any changes to this PR at this time. I still would like it merged at some point, but that will need to be done by someone from Microsoft and I haven't heard any feedback since I opened this about 1.5 years ago. If I hear back from someone there and it seems like they are willing to merge it, then I'll take the time to make updates.

**Comment by meche-gh** [comment]
Makes sense. This pull's assigned and I'm the review has pinged them already :) will give it a wait

**Review by zm-cttae — APPROVED** [review]
I'm wrong about the scope of this PR & I unfortunately read the last few messages of #14836 first.
I see that this pull is specifically for functionality `#1` - the "find" op.
Gonna see if I can dismiss my review and replace it with just the icon change.

File: src/vs/editor/contrib/find/findController.ts [hunk 1]
```diff
@@ -757,13 +760,54 @@ StartFindReplaceAction.addImplementation(0, (accessor: ServicesAccessor, editor:
 	});
 });
 
+export class FindAllInCurrentFileAction extends EditorAction {
```
[the remark this answers is not in the document]
Comment (meche-gh) [reply src/vs/editor/contrib/find/findController.ts:763]: Soo this pull is correct and is for the find op, hence why it's more complex and shells out to the search controller.

File: src/vs/editor/contrib/find/findWidget.ts [hunk 2]
```diff
@@ -47,6 +47,7 @@ export const findReplaceIcon = registerIcon('find-replace', Codicon.replace, nls
 export const findReplaceAllIcon = registerIcon('find-replace-all', Codicon.replaceAll, nls.localize('findReplaceAllIcon', 'Icon for \'Replace All\' in the editor find widget.'));
 export const findPreviousMatchIcon = registerIcon('find-previous-match', Codicon.arrowUp, nls.localize('findPreviousMatchIcon', 'Icon for \'Find Previous\' in the editor find widget.'));
 export const findNextMatchIcon = registerIcon('find-next-match', Codicon.arrowDown, nls.localize('findNextMatchIcon', 'Icon for \'Find Next\' in the editor find widget.'));
+export const findAllMatchesIcon = registerIcon('find-all-in-current-file', Codicon.search, nls.localize('findAllMatchesIcon', 'Icon for \'Find All in Current File\' in the editor find widget.'));
```
Comment (meche-gh) [inline src/vs/editor/contrib/find/findWidget.ts:50]:
```suggestion
export const findAllMatchesIcon = registerIcon('find-all-in-current-file', Codicon.newFile, nls.localize('findAllMatchesIcon', 'Icon for \'Find All in Current File\' in the editor find widget.'));
```

**Review by zm-cttae — COMMENTED** [review]
Right, here is the review with the icon change

File: src/vs/editor/contrib/find/findWidget.ts (hunk 2 above)
Comment (zm-cttae) [reply src/vs/editor/contrib/find/findWidget.ts:50]: @ssigwart could you merge this suggestion? It's the correct icon, matching the search pane next to the vertical Activity Bar.

Comment (ssigwart) [reply src/vs/editor/contrib/find/findWidget.ts:50]: @zm-cttae, I had taken a look at this when you first suggested, but this branch is so far behind main now and there were major merge conflicts. It's not worth my time fixing that without some type of feedback from Microsoft indicating this PR may be merged.

Comment (zm-cttae) [reply src/vs/editor/contrib/find/findWidget.ts:50]: It'd be possible to make the change and resolve the review.. looking to keep my lists clear

Comment (ssigwart) [reply src/vs/editor/contrib/find/findWidget.ts:50]: It doesn't seem that useful to change the icon when the whole thing won't work in its current state due to VS Code internal changes since I originally wrote this. I assume that Microsoft has other priorities and won't be looking at this, but I leave it open just in case.

**Comment by vivodi** [comment]
Is this still relevant?

**Comment by ssigwart** [comment]
I still think it's a viable solution to #14836, which is still open. However, it would need merge conflicts fixed first. I don't want to spend time doing that until someone from Microsoft indicates that this PR had some likelihood of being merged.
